frontend/src/metabase/search/components/filters/CreatedByFilter/CreatedByFilter.tsx

```diff
@@ -0,0 +1,19 @@
+import { t } from "ttag";
+import type { SearchSidebarFilterComponent } from "metabase/search/types";
+import { CreatedByDisplay } from "metabase/search/components/filters/CreatedByFilter/CreatedByDisplay";
+import { CreatedByContent } from "metabase/search/components/filters/CreatedByFilter/CreatedByContent";
+
+export const CreatedByFilter: SearchSidebarFilterComponent<"created_by"> = {
+  iconName: "person",
+  title: t`Creator`,
+  DisplayComponent: CreatedByDisplay,
+  ContentComponent: CreatedByContent,
+  fromUrl: value => {
+    if (!value || Array.isArray(value)) {
+      return undefined;
+    }
+    const numValue = Number(value);
+    return numValue && !isNaN(numValue) ? numValue : undefined;
```

Comment:
Ah we actually want `undefined` when userId=0 - users have IDs starting from 1. I'll also add a test case for negative numbers since anybody can put anything in the URL